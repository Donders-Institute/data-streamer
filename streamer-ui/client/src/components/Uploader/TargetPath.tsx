import React from "react";

interface IProps {
    isSelectedProject: boolean;
    projectNumber: string;
    isSelectedSubject: boolean;
    subjectLabel: string;
    isSelectedSession: boolean;
    isSelectedDataType: boolean;
    dataType: string;
    sessionLabel: string;
}

const TargetPath: React.FC<IProps> = (
    {
        isSelectedProject,
        projectNumber,
        isSelectedSubject,
        subjectLabel,
        isSelectedSession,
        isSelectedDataType,
        dataType,
        sessionLabel
    }) => {
    let forwardSlashPath = (
        <span style={{ fontWeight: "bold", color: "#f45709" }}>/</span>
    );
    let projectPath = (
        <span style={{ fontWeight: "bold", color: "#f45709" }}>project</span>
    );
    let rawPath = (
        <span style={{ fontWeight: "bold", color: "#f45709" }}>raw</span>
    );
    let subjectPath = (
        <span style={{ fontWeight: "bold", color: "#f45709" }}>sub-</span>
    );
    let sessionPath = (
        <span style={{ fontWeight: "bold", color: "#f45709" }}>ses-</span>
    );

    let projectNumberPath = (
        <span style={{ fontStyle: "italic" }}>(projectnumber)</span>
    );
    let subjectLabelPath = (
        <span style={{ fontStyle: "italic" }}>(subjectlabel)</span>
    );
    let dataTypePath = <span style={{ fontStyle: "italic" }}>(datatype)</span>;
    let sessionLabelPath = (
        <span style={{ fontStyle: "italic" }}>(sessionlabel)</span>
    );

    if (isSelectedProject) {
        projectNumberPath = (
            <span style={{ fontWeight: "bold", color: "#f45709" }}>
                {projectNumber}
            </span>
        );
    }
    if (isSelectedSubject) {
        subjectLabelPath = (
            <span style={{ fontWeight: "bold", color: "#f45709" }}>
                {subjectLabel}
            </span>
        );
    }
    if (isSelectedDataType && dataType !== "other") {
        dataTypePath = (
            <span style={{ fontWeight: "bold", color: "#f45709" }}>{dataType}</span>
        );
    }
    if (isSelectedDataType) {
        if (dataType === "other" || dataType === "") {
            dataTypePath = <span style={{ fontStyle: "italic" }}>(datatype)</span>;
        } else {
            dataTypePath = (
                <span style={{ fontWeight: "bold", color: "#f45709" }}>
                    {dataType}
                </span>
            );
        }
    }
    if (isSelectedSession) {
        sessionLabelPath = (
            <span style={{ fontWeight: "bold", color: "#f45709" }}>
                {sessionLabel}
            </span>
        );
    }

    return (
        <div>
            {forwardSlashPath}
            {projectPath}
            {forwardSlashPath}
            {projectNumberPath}
            {forwardSlashPath}
            {rawPath}
            {forwardSlashPath}
            {subjectPath}
            {subjectLabelPath}
            {forwardSlashPath}
            {sessionPath}
            {sessionLabelPath}
            {forwardSlashPath}
            {dataTypePath}
            {forwardSlashPath}
        </div>
    );
};

export default TargetPath;
