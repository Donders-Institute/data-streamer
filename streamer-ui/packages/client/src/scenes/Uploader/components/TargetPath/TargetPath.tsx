import React from "react";

import { Tooltip } from "antd";

interface TargetPathProps {
    isSelectedProject: boolean;
    isSelectedSubject: boolean;
    isSelectedSession: boolean;
    isSelectedDataType: boolean;
    projectNumber: string;
    subjectLabel: string;
    dataType: string;
    sessionLabel: string;
}

const TargetPath: React.FC<TargetPathProps> = (
    {
        isSelectedProject,
        isSelectedSubject,
        isSelectedSession,
        isSelectedDataType,
        projectNumber,
        subjectLabel,
        dataType,
        sessionLabel
    }) => {
    let drivePath = (
        <span style={{ fontWeight: "bold", color: "#52c41a" }}>P:</span>
    );
    let backwardSlashPath = (
        <span style={{ fontWeight: "bold", color: "#52c41a" }}>\</span>
    );
    let rawPath = (
        <span style={{ fontWeight: "bold", color: "#52c41a" }}>raw</span>
    );
    let subjectPath = (
        <span style={{ fontWeight: "bold", color: "#52c41a" }}>sub-</span>
    );
    let sessionPath = (
        <span style={{ fontWeight: "bold", color: "#52c41a" }}>ses-</span>
    );

    let projectNumberPath = (
        <span>(projectnumber)</span>
    );
    let subjectLabelPath = (
        <span>(subjectlabel)</span>
    );
    let dataTypePath = <span >(datatype)</span>;
    let sessionLabelPath = (
        <span>(sessionlabel)</span>
    );

    if (isSelectedProject) {
        projectNumberPath = (
            <span style={{ fontWeight: "bold", color: "#52c41a" }}>
                {projectNumber}
            </span>
        );
    }

    if (isSelectedSubject) {
        subjectLabelPath = (
            <span style={{ fontWeight: "bold", color: "#52c41a" }}>
                {subjectLabel}
            </span>
        );
    }

    if (isSelectedSession) {
        sessionLabelPath = (
            <span style={{ fontWeight: "bold", color: "#52c41a" }}>
                {sessionLabel}
            </span>
        );
    }

    if (isSelectedDataType) {
        dataTypePath = (
            <span style={{ fontWeight: "bold", color: "#52c41a" }}>{dataType}</span>
        );

        if (dataType === "other" || dataType === "") {
            dataTypePath = <span>(datatype)</span>;
        } else {
            dataTypePath = (
                <span style={{ fontWeight: "bold", color: "#52c41a" }}>
                    {dataType}
                </span>
            );
        }
    }

    return (
        <React.Fragment>
            <Tooltip placement="bottomLeft" title="Destination folder where files are uploaded to">
                {drivePath}
                {backwardSlashPath}
                {projectNumberPath}
                {backwardSlashPath}
                {rawPath}
                {backwardSlashPath}
                {subjectPath}
                {subjectLabelPath}
                {backwardSlashPath}
                {sessionPath}
                {sessionLabelPath}
                {backwardSlashPath}
                {dataTypePath}
                {backwardSlashPath}
            </Tooltip>
        </React.Fragment>
    );
};

export default TargetPath;
