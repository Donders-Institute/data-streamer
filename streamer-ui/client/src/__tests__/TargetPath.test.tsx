import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import TargetPath from "../components/Uploader/TargetPath";

describe("<TargetPath />", () => {
    it("renders initial targetpath component", () => {
        const props = {
            isSelectedProject: false,
            projectNumber: "",
            isSelectedSubject: false,
            subjectLabel: "",
            isSelectedSession: false,
            isSelectedDataType: false,
            dataType: "",
            sessionLabel: ""
        };
        const wrapper = shallow(<TargetPath {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders targetpath component with project selected", () => {
        const props = {
            isSelectedProject: true,
            projectNumber: "12345678.01",
            isSelectedSubject: false,
            subjectLabel: "",
            isSelectedSession: false,
            sessionLabel: "",
            isSelectedDataType: false,
            dataType: ""
        };
        const wrapper = shallow(<TargetPath {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders targetpath component with subject selected", () => {
        const props = {
            isSelectedProject: true,
            projectNumber: "12345678.01",
            isSelectedSubject: true,
            subjectLabel: "1",
            isSelectedSession: false,
            sessionLabel: "",
            isSelectedDataType: false,
            dataType: ""
        };
        const wrapper = shallow(<TargetPath {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders targetpath component with session selected", () => {
        const props = {
            isSelectedProject: true,
            projectNumber: "12345678.01",
            isSelectedSubject: true,
            subjectLabel: "1",
            isSelectedSession: true,
            sessionLabel: "2",
            isSelectedDataType: false,
            dataType: ""
        };
        const wrapper = shallow(<TargetPath {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders targetpath component with data type other", () => {
        const props = {
            isSelectedProject: true,
            projectNumber: "12345678.01",
            isSelectedSubject: true,
            subjectLabel: "1",
            isSelectedSession: true,
            sessionLabel: "2",
            isSelectedDataType: true,
            dataType: "other"
        };
        const wrapper = shallow(<TargetPath {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders targetpath component with data type", () => {
        const props = {
            isSelectedProject: true,
            projectNumber: "12345678.01",
            isSelectedSubject: true,
            subjectLabel: "1",
            isSelectedSession: true,
            sessionLabel: "2",
            isSelectedDataType: true,
            dataType: "mri"
        };
        const wrapper = shallow(<TargetPath {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});