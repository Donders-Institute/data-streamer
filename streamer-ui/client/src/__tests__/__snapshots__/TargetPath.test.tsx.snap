// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TargetPath /> renders initial targetpath component 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    project
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (projectnumber)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    raw
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    sub-
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (subjectlabel)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    ses-
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (sessionlabel)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (datatype)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
</div>
`;

exports[`<TargetPath /> renders targetpath component with data type 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    project
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    12345678.01
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    raw
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    sub-
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    1
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    ses-
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    2
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    mri
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
</div>
`;

exports[`<TargetPath /> renders targetpath component with data type other 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    project
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    12345678.01
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    raw
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    sub-
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    1
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    ses-
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    2
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (datatype)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
</div>
`;

exports[`<TargetPath /> renders targetpath component with project selected 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    project
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    12345678.01
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    raw
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    sub-
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (subjectlabel)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    ses-
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (sessionlabel)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (datatype)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
</div>
`;

exports[`<TargetPath /> renders targetpath component with session selected 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    project
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    12345678.01
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    raw
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    sub-
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    1
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    ses-
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    2
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (datatype)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
</div>
`;

exports[`<TargetPath /> renders targetpath component with subject selected 1`] = `
<div>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    project
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    12345678.01
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    raw
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    sub-
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    1
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    ses-
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (sessionlabel)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
  <span
    style={
      Object {
        "fontStyle": "italic",
      }
    }
  >
    (datatype)
  </span>
  <span
    style={
      Object {
        "color": "#f45709",
        "fontWeight": "bold",
      }
    }
  >
    /
  </span>
</div>
`;
